### PR TITLE
[System.Data] referencesource -> corefx: ProviderBase & SqlClient stuff

### PR DESCRIPTION
**Pull request by EgorBo**
Also, Resx2cs is now able to wrap SR into a namespace (for example, 'namespace System'). It's needed because some corefx stuff expects it to be there, for example:
`using Res = System.SR;`

### CODE REVIEW

File: mcs/class/System.Data/corefx.common.sources [hunk 1]
```diff
@@ -228,41 +236,39 @@
...
-
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/Common/ActivityCorrelator.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPoolProviderInfo.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/DbReferenceCollection.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/ProviderBase/TimeoutTimer.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/SqlClient/ApplicationIntent.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlError.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlErrorCollection.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlInfoMessageEvent.cs
+../../../external/corefx/src/System.Data.SqlClient/src/System\Data\OperationAbortedException.cs
```
Comment (rjvdboon) [inline mcs/class/System.Data/corefx.common.sources]: You probably meant forward slashes here

Comment (EgorBo) [inline mcs/class/System.Data/corefx.common.sources]: @rjvdboon good catch! thanks

**Comment by EgorBo** [comment]
@marek-safar 
After 'bump corefx' I removed all SR changes in my PR, it's now ready to be merged.

**Comment by JustArchi** [comment]
Most likely caused pretty serious regression - https://bugzilla.xamarin.com/show_bug.cgi?id=52795

**Comment by EgorBo** [comment]
@JustAchi thanks for the repro, I am investigating this now.

**Comment by JustArchi** [comment]
@EgorBo Any progress? I'm not in a hurry (especially because it's dev branch) but I'd like to keep helping with testing latest master with my projects, and 90% of them are currently running on 15 feb release, since I can't really do anything with broken MySQL 😢

**Comment by EgorBo** [comment]
@JustArchi ah, sorry, I've just fixed it. I am working on integration tests including this case to avoid such issues in future (while migrating to corefx).

**Pull request merged**